### PR TITLE
Migrate CI from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by tnir**
Signed-off-by: Takuya Noguchi <takninnovationresearch@gmail.com>

RIP Travis CI.

- Ruby versions: unchanged (with revision)
- Allow failure: add for Rails head with Ruby 2.7 and 3.0 besides for Ruby head
- Gem caching: defer to `ruby/setup-ruby@v1` actions
- Badge: updated

rails/webpacker#2614 etc.

### CODE REVIEW

**Comment by tnir** [comment]
Actions executed: https://github.com/tnir/jbuilder/actions/runs/1118808623

**Comment by tnir** [comment]
@kaspth What do you think?

**Comment by tnir** [comment]
@yuki24 What do you think?

**Comment by tnir** [comment]
Who can merge this?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 Contributing to Jbuilder
 =====================
 
-[![Build Status](https://api.travis-ci.org/rails/jbuilder.svg?branch=master)][travis]
+[![Build Status](https://github.com/rails/jbuilder/workflows/Ruby%20test/badge.svg)][test]
 [![Gem Version](https://badge.fury.io/rb/jbuilder.svg)][gem]
 [![Code Climate](https://codeclimate.com/github/rails/jbuilder/badges/gpa.svg)][codeclimate]
 
-[travis]: https://travis-ci.org/rails/jbuilder
+[test]: https://github.com/rails/webpacker/actions?query=branch%3Amaster
```
Comment (kangkyu) [inline CONTRIBUTING.md:8]: it shows `webpacker` this line. possibly `jbuilder` I'm guessing

Comment (tnir) [inline CONTRIBUTING.md:8]: Right, it should be as follows:
```suggestion
[test]: https://github.com/rails/jbuilder/actions?query=branch%3Amaster
```

Comment (tnir) [reply CONTRIBUTING.md]: Applied in b1894ef.

Comment (tnir) [reply CONTRIBUTING.md]: Fixed in b1894ef.

**Pull request merged**